client/src/app/rotation-options/rotation-options.component.html

```diff
@@ -0,0 +1,44 @@
+<div class="rotation-text"
+     *ngIf="dpsService.selfJob === jobConsts.SCHOLAR.name">
+  <span class="rotation-tag">Rotation: </span>
+  <span>
+  	{{dpsService.rotation.energyDrain}} Energy Drain,
+  	{{dpsService.rotation.adloquium}} Adloquium,
+  	{{dpsService.rotation.succor}} Succor,
+  	{{dpsService.rotation.ressurection}} Ressurection per min
+  </span>
+  <i class="edit-icon" nz-icon [nzType]="'edit'" (click)="editRotation()"></i>
+</div>
+<nz-modal #editModal [(nzVisible)]="modalVisible"
+    (nzOnCancel)="handleCancel()" (nzOnOk)="handleConfirm()">
+  <ng-container *nzModalContent>
+    <div class="option-edit-container">
+      <nz-input-number class="option-edit-input"
+          [(ngModel)]="editCache.energyDrain"
+          [nzStep]="1" [nzSize]="'small'">
```

Comment:
can we get some input validation for these? right now it will happily let me put in invalid values (> 4 for energy drain), < 0 for anything else.

the backend server obviously should check for input validity on requests too, but i think having a check before any request is sent and a visual indication that something is wrong would be helpful.

i'm open to discussion about this though, and also open to discussing how tryhard the checks should be. energy drain between 0 and 4, and all other values >= 0 are relatively simple checks. but in theory we could also check that sum of the number of gcd heals/rez is not greater than the smallest number of gcds per minute from any gearset in the table.

Reply:
also it lets me put decimals and that's also a big nope

Reply:
Changed the component, since the library we're using conveniently has one just for numeric inputs and support for validation.  I can't get the screen reader to read out what the field is for when I tab over to it, but it's already labelled, so that's probably fine?